### ISSUE TITLE
Bastion host deployment keeps failing in IAD.

### ISSUE THREAD

**Issue by Luna-LHY** [aws-ia/cfn-ps-linux-bastion#103]
Hi team, we use this `cfn-ps-linux-bastion` repository as submodule in our project and deploy our bastion host accordingly. However, after a recent update of Amazon Linux2 AMI in IAD, our bastion host deployment started to fail in this region. After checking `cfn-init.log`, looks like the root cause is that the latest AMI in IAD doesn't have `auditd` installed by default. But we are trying to restart auditd service in [auditing_configure.sh](https://github.com/aws-ia/cfn-ps-linux-bastion/blob/4363a4c96109f6f28fc8adfc627e06027a7a4aba/scripts/auditing_configure.sh#L17), which resulted in `Failed to restart auditd.service: Unit not found.`

cfn-init.log:
```
04:20:26,514 [DEBUG] Command a-add_auditd_rules output: + case "${BASTION_OS}" in
+ add_the_rules
+ cat /tmp/auditd.rules
+ rm /tmp/auditd.rules
+ restart_services
+ case "${BASTION_OS}" in
+ /usr/sbin/service auditd restart
Redirecting to /bin/systemctl restart auditd.service
Failed to restart auditd.service: Unit not found.

2024-04-01 04:20:26,514 [ERROR] Error encountered during build of config: Command a-add_auditd_rules failed
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/cfnbootstrap/construction.py", line 579, in run_config
    CloudFormationCarpenter(config, self._auth_config, self.strict_mode).build(worklog)
  File "/usr/lib/python3.7/site-packages/cfnbootstrap/construction.py", line 278, in build
    self._config.commands)
  File "/usr/lib/python3.7/site-packages/cfnbootstrap/command_tool.py", line 127, in apply
    raise ToolError(u"Command %s failed" % name)
cfnbootstrap.construction_errors.ToolError: Command a-add_auditd_rules failed
2024-04-01 04:20:26,517 [ERROR] -----------------------BUILD FAILED!------------------------
2024-04-01 04:20:26,517 [ERROR] Unhandled exception during build: Command a-add_auditd_rules failed
Traceback (most recent call last):
  File "/usr/bin/cfn-init", line 181, in <module>
    worklog.build(metadata, configSets, strict_mode)
  File "/usr/lib/python3.7/site-packages/cfnbootstrap/construction.py", line 137, in build
    Contractor(metadata, strict_mode).build(configSets, self)
  File "/usr/lib/python3.7/site-packages/cfnbootstrap/construction.py", line 567, in build
    shttps://aws.amazon.com/cn/codebuild/features/?nc=sn&loc=2elf.run_config(config, worklog)
  File "/usr/lib/python3.7/site-packages/cfnbootstrap/construction.py", line 579, in run_config
    CloudFormationCarpenter(config, self._auth_config, self.strict_mode).build(worklog)
  File "/usr/lib/python3.7/site-packages/cfnbootstrap/construction.py", line 278, in build
    self._config.commands)
  File "/usr/lib/python3.7/site-packages/cfnbootstrap/command_tool.py", line 127, in apply
    raise ToolError(u"Command %s failed" % name)
cfnbootstrap.construction_errors.ToolError: Command a-add_auditd_rules failed
2024-04-01 04:20:26,766 [DEBUG] CloudFormation client initialized with endpoint https://cloudformation.us-east-1.amazonaws.com
2024-04-01 04:20:26,766 [DEBUG] Signaling resource BastionAutoScalingGroup in stack clickhouse-cluster-debug-BastionStack-2OMZEZYS7CVN with unique ID i-0537dfd1055d445ba and status FAILURE
```

Looks like `auditd` is only missing in IAD, as I deployed AL2 AMI in other regions and I was able to find `auditd` under `/usr/sbin` path. Is this expected? If not, can we fix the logic in auditing_configure.sh and exclude IAD region so that it won't block deployments in IAD?

**Comment by troy-ameigh** [comment]
@Luna-LHY 

Have you tried this again. We have not been able to duplicate.

**Comment by Luna-LHY** [comment]
> @Luna-LHY
> 
> Have you tried this again. We have not been able to duplicate.

Yeah I tried to deploy [our project](https://github.com/aws-ia/cfn-ps-clickhouse-cluster) in IAD quite a few times, and bastion host failed to deploy with the same error.

However, I found one thing very interesting...I also tested bastion host deployment by using this solution https://aws.amazon.com/solutions/implementations/linux-bastion/?nc1=h_ls, which I believe uses the same code in [cfn-ps-linux-bastion](https://github.com/aws-ia/cfn-ps-linux-bastion) repo for bastion deployment. And deployment succeeded with:

```
2024-04-15 01:12:08,205 [DEBUG] Command a-add_auditd_rules output: + case "${BASTION_OS}" in
+ add_the_rules
+ cat /tmp/auditd.rules
+ rm /tmp/auditd.rules
+ restart_services
+ case "${BASTION_OS}" in
+ /usr/sbin/service auditd restart
Stopping logging: [  OK  ]^M
Redirecting start to /bin/systemctl start auditd.service

2024-04-15 01:12:08,205 [DEBUG] Running command b-bootstrap
2024-04-15 01:12:08,205 [DEBUG] No test for command b-bootstrap
osrelease ended
```
So it seems to me it was able to run`/usr/sbin/service auditd restart` in the script. However, when I try to check audit.d status with `sudo service auditd status` or `sudo /bin/systemctl status auditd.service`, I'm still getting `Unit auditd.service could not be found`....It looks like auditd service doesn't exist, which is similar to what I found when deploying our project. But the difference is that this time it didn't return any error. Why is this happening?
